kedro/io/hdf_s3.py

```diff
@@ -103,30 +106,26 @@ def __init__(
                 attribute is None, save version will be autogenerated.
 
         """
-        _credentials = deepcopy(credentials) or {}
+        _credentials = copy.deepcopy(credentials) or {}
         _s3 = S3FileSystem(client_kwargs=_credentials)
         super().__init__(
             PurePosixPath("{}/{}".format(bucket_name, filepath)),
             version,
             exists_function=_s3.exists,
             glob_function=partial(_s3.glob, refresh=True),
         )
-
-        default_load_args = {}  # type: Dict[str, Any]
-        default_save_args = {}  # type: Dict[str, Any]
-        self._key = key
         self._bucket_name = bucket_name
+        self._key = key
         self._credentials = _credentials
-        self._load_args = (
-            {**default_load_args, **load_args}
-            if load_args is not None
-            else default_load_args
-        )
-        self._save_args = (
-            {**default_load_args, **save_args}
-            if save_args is not None
-            else default_save_args
-        )
+
+        # Handle default load and save arguments
```

Comment:
Why this (and all other datasets in `kedro.io` that are eligible) are not inheriting from the new `MixIn`? (maybe I missed something here, sorry about that!)

The code feels inconsistent now (+ all this duplication can go away!)

Reply:
@tsanikgr https://github.com/quantumblacklabs/kedro/issues/14#issuecomment-507703582

Reply:
OK thanks! @idanov, since the mix in was introduced, and is solving the problem "of the wrong abstraction", should we just leverage it in the core datasets as well?

Reply:
@idanov @tsanikgr @tolomea Let me know what you all decide as the core team here; I’m OK with it as is (a current marginal improvement to `core` with potential to move the mix-in from `contrib` after proving value in the future) or pushing the mix-in to `core` now. I’m more keen on merging to `develop` sooner than later due to the number of datasets touched, if possible. :)

Reply:
I think the current version is good for now. Thanks @deepyaman for accommodating for all the comments!